create_derivatives/tests.py

```diff
@@ -118,3 +142,42 @@ def test_run(self, mock_get_object, mock_init):
     def tearDown(self):
         for f in Path(settings.SRC_DIR).iterdir():
             f.unlink()
+
+
+class AWSUploadTestCase(TestCase):
+
+    def setUp(self):
+        """Creates a temp directory and adds an uncompressed bag fixture to the temp directory and database"""
+        path = Path(settings.TMP_DIR)
+        if not path.exists():
+            path.mkdir(parents=True)
+        bag = "3aai9usY3AZzCSFkB3RSQ9"
+        fixture_directory = "aws_upload_bag"
+        bag_path = Path(settings.TMP_DIR, bag)
+        if not Path(bag_path).exists():
+            shutil.copytree(Path("create_derivatives", "fixtures", fixture_directory, bag), bag_path)
+            Bag.objects.create(
+                bag_identifier="sdfjldskj",
+                bag_path=bag_path,
+                origin="digitization",
+                as_data="sdjfkldsjf",
+                dimes_identifier=bag,
+                process_status=Bag.MANIFESTS_CREATED)
+
+    @patch("create_derivatives.clients.AWSClient.__init__")
+    @patch("create_derivatives.clients.AWSClient.upload_files")
+    def test_run(self, mock_upload_files, mock_init):
```

Comment:
I think it might also be good to check how many times the upload mock was called. This test doesn't really touch the AWSClient, which is fine, but we want to make sure that `upload_files` method gets called.